src/Lean/Server/FileWorker.lean

```diff
@@ -181,65 +193,97 @@ section Initialization
     let stderr ← IO.asTask (processStderr "") Task.Priority.dedicated
     let stdout := String.trim (← lakeProc.stdout.readToEnd)
     let stderr ← IO.ofExcept stderr.get
-    match (← lakeProc.wait) with
+    let exitCode ← lakeProc.wait
+    return ⟨spawnArgs, exitCode, stdout, stderr⟩
+
+  /-- Use `lake print-paths` to compile dependencies on the fly and add them to `LEAN_PATH`.
+  Compilation progress is reported to `hOut` via LSP notifications. Return the search path for
+  source files. -/
+  partial def lakeSetupSearchPath (lakePath : System.FilePath) (m : DocumentMeta) (imports : Array Import) (hOut : FS.Stream) : IO SearchPath := do
+    let result ← runLakePrintPaths lakePath m imports hOut
+    let cmdStr := " ".intercalate (toString result.spawnArgs.cmd :: result.spawnArgs.args.toList)
+    match result.exitCode with
     | 0 =>
-      let Except.ok (paths : LeanPaths) ← pure (Json.parse stdout >>= fromJson?)
-        | throwServerError s!"invalid output from `{cmdStr}`:\n{stdout}\nstderr:\n{stderr}"
+      let Except.ok (paths : LeanPaths) ← pure (Json.parse result.stdout >>= fromJson?)
+        | throwServerError s!"invalid output from `{cmdStr}`:\n{result.stdout}\nstderr:\n{result.stderr}"
       initSearchPath (← getBuildDir) paths.oleanPath
       paths.loadDynlibPaths.forM loadDynlib
       paths.srcPath.mapM realPathNormalized
     | 2 => pure []  -- no lakefile.lean
     -- error from `--no-build`
-    | 3 => throwServerError s!"Imports are out of date and must be rebuilt; use the \"Restart File\" command in your editor.\n\n{stdout}"
-    | _ => throwServerError s!"`{cmdStr}` failed:\n{stdout}\nstderr:\n{stderr}"
+    | 3 => throwServerError s!"Imports are out of date and must be rebuilt; use the \"Restart File\" command in your editor.\n\n{result.stdout}"
+    | _ => throwServerError s!"`{cmdStr}` failed:\n{result.stdout}\nstderr:\n{result.stderr}"
+
+  def setupSrcSearchPath (m : DocumentMeta) (hOut : FS.Stream) (headerStx : Syntax) : IO SearchPath := do
+    let some path := System.Uri.fileUriToPath? m.uri
+      | return ← initSrcSearchPath
+
+    -- NOTE: we assume for now that `lakefile.lean` does not have any non-stdlib deps
```

Comment:
Suggest using "Lean" or "core Lean" rather than "stdlib", which should refer to Std.

Reply:
Agreed. By the way, this comment is two years old at this point :)